pkg/k8s/apis/cilium.io/v2alpha1/bgp_cluster_types.go

```diff
@@ -108,9 +108,13 @@ type CiliumBGPPeer struct {
 	// PeerASN is the ASN of the peer BGP router.
 	// Supports extended 32bit ASNs.
 	//
+	// If peerASN is 0, the ASN negotiation will be disabled when bringing up
```

Comment:
`the ASN negotiation will be disabled` is GoBGP's internal term and not that clear. I would suggest `ASN will be determined based on the peer's OPEN message`.

https://github.com/osrg/gobgp/blob/master/pkg/server/fsm.go#L1339-L1362

Reply:
Maybe ASN validation would be better term. We are explicitly disabling ASN validation by setting it to 0. Validation happens [here](https://github.com/osrg/gobgp/blob/master/pkg/packet/bgp/validate.go#L348-L350). [RFC](https://www.ietf.org/rfc/rfc4271.txt) is also explicit about it 

`If the Autonomous System field of the OPEN message is unacceptable,
   then the Error Subcode MUST be set to Bad Peer AS.  The determination
   of acceptable Autonomous System numbers is outside the scope of this
   protocol. `

There are two consequences of setting ASN to 0; 1. ASN validation is disabled, 2. PeerASN is detected using Open message received from Peer. I can rephrase it like this 

`If peerASN is 0, the BGP OPEN message validation of ASN will be disabled and ASN will be determined based on peer's OPEN message.` WDYT?